apps/src/code-studio/progress.js

```diff
@@ -81,72 +85,16 @@ progress.renderCourseProgress = function (scriptData, currentLevelId) {
   );
 };
 
-// Return the level with the highest progress, or the first level if none have
-// been attempted
-progress.bestResultLevelId = function (levelIds, progressData) {
-  // The usual case
-  if (levelIds.length === 1) {
-    return levelIds[0];
-  }
-
-  // Return the level with the highest result
-  var attemptedIds = levelIds.filter(id => progressData[id]);
-  if (attemptedIds.length === 0) {
-    // None of them have been attempted, just return the first
-    return levelIds[0];
-  }
-  var bestId = attemptedIds[0];
-  var bestResult = progressData[bestId];
-  attemptedIds.forEach(function (id) {
-    var result = progressData[id];
-    if (result > bestResult) {
-      bestId = id;
-      bestResult = result;
-    }
-  });
-  return bestId;
-};
-
-function loadProgress(scriptData, currentLevelId, saveAnswersBeforeNavigation = false) {
-
-  let store = createStore((state = {}, action) => {
-    if (action.type === 'MERGE_PROGRESS') {
-      // TODO: _.mergeWith after upgrading to Lodash 4+
-      let newProgress = {};
-      Object.keys(Object.assign({}, state.progress, action.progress)).forEach(key => {
-        newProgress[key] = mergeActivityResult(state.progress[key], action.progress[key]);
-      });
-
-      const stages = state.stages.map(stage => Object.assign({}, stage, {levels: stage.levels.map((level, index) => {
-        const id = level.uid || progress.bestResultLevelId(level.ids, newProgress);
-
-        if (action.peerReviewsPerformed && stage.flex_category === 'Peer Review') {
-          Object.assign(level, action.peerReviewsPerformed[index]);
-        }
-
-        return Object.assign({}, level, {
-          status: level.kind === 'peer_review' ? level.status : activityCssClass(newProgress[id]),
-          id: id,
-          url: level.url
-        });
-      })}));
-
-      return Object.assign({}, state, {
-        progress: newProgress,
-        stages: stages
-      });
-    } else if (action.type === 'UPDATE_FOCUS_AREAS') {
-      return Object.assign({}, state, {
-        changeFocusAreaPath: action.changeFocusAreaPath,
-        focusAreaPositions: action.focusAreaPositions
-      });
-    } else if (action.type === 'SHOW_LESSON_PLAN_LINKS') {
-      return Object.assign({}, state, {
-        showLessonPlanLinks: true
-      });
-    }
-    return state;
-  }, {
+/**
+ * Creates a redux store with our initial progress
+ * @param scriptData
+ * @param currentLevelId
+ * @param saveAnswersBeforeNavigation
+ * @returns {object} The created redux store
+ */
+function createStoreWithProgress(scriptData, currentLevelId,
```

Comment:
renamed method to be more descriptive. Moved all of the reducers into progressRedux